cartoframes/viz/kuviz.py

```diff
@@ -36,45 +38,49 @@ def is_public(self):
 
     def get_layers(self, maps_api_key='default_public'):
         for layer in self._layers:
-            layer.source.dataset.credentials = self._context
+            layer.source.dataset.credentials = self._credentials
 
             layer.source.credentials = {
-                'username': self._context.creds.username,
+                'username': self._credentials.username,
                 'api_key': maps_api_key,
-                'base_url': self._context.creds.base_url
+                'base_url': self._credentials.base_url
             }
 
         return self._layers
 
-    def sync_layers(self, table_name, context=None):
+    def sync_layers(self, table_name, credentials=None):
         for idx, layer in enumerate(self._layers):
             table_name = normalize_name("{name}_{idx}".format(name=table_name, idx=idx + 1))
 
-            from ..auth import _default_context
-            dataset_context = context or layer.source.dataset.credentials or _default_context
+            from ..auth import _default_credentials
+            dataset_credentials = credentials or layer.source.dataset.credentials or _default_credentials
 
-            self._sync_layer(layer, table_name, dataset_context)
+            self._sync_layer(layer, table_name, dataset_credentials)
 
-    def _sync_layer(self, layer, table_name, context):
+    def _sync_layer(self, layer, table_name, credentials):
         if not layer.source.dataset.is_saved_in_carto:
-            layer.source.dataset.upload(table_name=table_name, credentials=context)
-            layer.source = Source(table_name, context=context)
+            layer.source.dataset.upload(table_name=table_name, credentials=credentials)
+            layer.source = Source(table_name, credentials=credentials)
             warn('Table `{}` created. In order to publish the map, you will need to create a new Regular API '
                  'key with permissions to Maps API and the table `{}`. You can do it from your CARTO dashboard or '
                  'using the Auth API. You can get more info at '
                  'https://carto.com/developers/auth-api/guides/types-of-API-Keys/'.format(table_name, table_name))
 
 
-def _create_kuviz(html, name, context=None, password=None):
-    km = _get_kuviz_manager(context)
+def _create_kuviz(html, name, credentials=None, password=None):
```

Comment:
I was thinking about that using the context, but showing this solution, I think it has more sense of creating the object here. Only a suggestion: we could avoid creating a new object in every `publish` call, maybe setting a new property in the Kuviz class:

```
def __init__(self, vmap, credentials=None):
    self._layers = deepcopy(vmap.layers)
    self._credentials = credentials
    self._auth_client = self._create_auth_client()

def _create_auth_client(self):
    return APIKeyAuthClient(
        base_url=self._credentials.base_url,
        api_key=self._credentials.api_key,
        session=self._credentials.session,
        client_id='cartoframes_{}'.format(__version__),
        user_agent='cartoframes_{}'.format(__version__)
    )
```

Reply:
:+1: I'll add this to the `set_credentials` method

Reply:
Top!